core/src/main/java/org/elasticsearch/index/mapper/RangeFieldMapper.java

```diff
@@ -427,10 +432,73 @@ protected void doXContentBody(XContentBuilder builder, boolean includeDefaults,
 
     /** Enum defining the type of range */
     public enum RangeType {
+        IP("ip_range") {
+            @Override
+            public Field getRangeField(String name, Range r) {
+                return new InetAddressRange(name, (InetAddress)r.from, (InetAddress)r.to);
+            }
+            @Override
+            public InetAddress parseFrom(RangeFieldType fieldType, XContentParser parser, boolean coerce, boolean included)
+                throws IOException {
+                InetAddress address = InetAddresses.forString(parser.text());
+                return included ? address : nextUp(address);
+            }
+            @Override
+            public InetAddress parseTo(RangeFieldType fieldType, XContentParser parser, boolean coerce, boolean included)
+                throws IOException {
+                InetAddress address = InetAddresses.forString(parser.text());
+                return included ? address : nextDown(address);
+            }
+            @Override
+            public InetAddress parse(Object value, boolean coerce) {
+                try {
+                    return value instanceof InetAddress ? (InetAddress) value : InetAddress.getByName((String) value);
```

Comment:
can you use `InetAddresses.forString` intead, which guarantees it won't do a dns lookup